src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+        vec1 = D_80AF19F8;
+        Matrix_MultVec3f(&vec1, &vec2);
+        actor->posRot2.pos.x = vec2.x;
+        actor->posRot2.pos.y = vec2.y;
+        actor->posRot2.pos.z = vec2.z;
+        actor->posRot2.rot.x = actor->posRot.rot.x;
+        actor->posRot2.rot.y = actor->posRot.rot.y;
+        actor->posRot2.rot.z = actor->posRot.rot.z;
+    }
+}
+
+void func_80AF03F4(EnRu1* this, GlobalContext* globalCtx) {
+}
+
+void func_80AF0400(EnRu1* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 temp = this->unk_25C;
+    s32 addr1 = D_80AF0858[temp];
+    s16 temp2 = this->unk_260;
+    SkelAnime* skelAnime = &this->skelAnime;
+    s32 addr2 = D_80AF0870[temp2];
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[5];
+
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_ru1.c", 1282);
+    func_80093D18(globalCtx->state.gfxCtx);
+
+    gSPSegment(gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(addr1));
+    gSPSegment(gfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr1));
+    gSPSegment(gfxCtx->polyOpa.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr2));
+    gDPSetEnvColor(gfxCtx->polyOpa.p++, 0x00, 0x00, 0x00, 0xFF);
+    gSPSegment(gfxCtx->polyOpa.p++, 0x0C, &D_80116280[2]);
+
+    gfxCtx->polyOpa.p =
+        SkelAnime_DrawSV2(globalCtx, skelAnime->skeleton, skelAnime->actorDrawTbl, skelAnime->dListCount, func_80AF02E8,
+                          func_80AF0368, &this->actor, gfxCtx->polyOpa.p);
+
+    Graph_CloseDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_ru1.c", 1309);
+}
+
+void func_80AF05D4(EnRu1* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 temp = this->unk_25C;
+    s32 addr1 = D_80AF0858[temp];
+    s16 temp2 = this->unk_260;
+    SkelAnime* skelAnime = &this->skelAnime;
+    s32 addr2 = D_80AF0870[temp2];
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    Gfx* dispRefs[5];
+
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_ru1.c", 1324);
+    func_80093D84(globalCtx->state.gfxCtx);
+
+    gSPSegment(gfxCtx->polyXlu.p++, 0x08, SEGMENTED_TO_VIRTUAL(addr1));
+    gSPSegment(gfxCtx->polyXlu.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr1));
+    gSPSegment(gfxCtx->polyXlu.p++, 0x09, SEGMENTED_TO_VIRTUAL(addr2));
+    gDPSetEnvColor(gfxCtx->polyXlu.p++, 0x00, 0x00, 0x00, this->unk_2A8);
+    gSPSegment(gfxCtx->polyXlu.p++, 0x0C, &D_80116280[0]);
+
+    gfxCtx->polyXlu.p = SkelAnime_DrawSV2(globalCtx, skelAnime->skeleton, skelAnime->actorDrawTbl,
+                                          skelAnime->dListCount, func_80AF02E8, NULL, &this->actor, gfxCtx->polyXlu.p);
+
+    Graph_CloseDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_ru1.c", 1353);
+}
+
+void EnRu1_Draw(EnRu1* this, GlobalContext* globalCtx) {
+    if (this->drawConfig < 0 || this->drawConfig >= 3 || D_80AF1A04[this->drawConfig] == 0) {
+        osSyncPrintf(VT_FGCOL(RED) "描画モードがおかしい!!!!!!!!!!!!!!!!!!!!!!!!!\n" VT_RST);
+        return;
+    }
+    D_80AF1A04[this->drawConfig](this, globalCtx);
```

Comment:
if you go with early return in update do it here too. I think both work, but just make sure to keep them consistent whichever you do pick